sdk/core/Azure.Core/src/DynamicData/AllowListConverterFactory.cs

```diff
@@ -0,0 +1,196 @@
+            {
+                if (!property.CanRead)
+                {
+                    return false;
+                }
+
+                if (!property.CanWrite && !IsAnonymousType(type))
+                {
+                    return false;
+                }
+
+                if (IsAllowedKnownType(property.PropertyType))
+                {
+                    continue;
+                }
+
+                // Trust but verify
+                if (ancestorTypes.Contains(property.PropertyType))
+                {
+                    continue;
+                }
+
+                // Recurse
+                ancestorTypes.Add(type);
+                if (!IsAllowedPoco(property.PropertyType, ancestorTypes))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        private static bool HasPublicParameterlessConstructor(Type type)
+        {
+            return type.GetConstructor(Type.EmptyTypes) != null;
+        }
+
+        private static bool IsAnonymousType(Type type)
+        {
+            return
+                type.Namespace == null &&
+                Attribute.IsDefined(type, typeof(CompilerGeneratedAttribute), false);
+        }
+
+        private static bool IsAllowedKnownType(Type type)
+        {
+            return IsAllowedPrimitive(type) ||
+                IsAllowedArray(type) ||
+                IsAllowedCollection(type) ||
+                IsAllowedInterface(type) ||
+
+                // TODO: separate out non-primitive values?
+                type == typeof(JsonElement) ||
+                type == typeof(JsonDocument) ||
+                type == typeof(MutableJsonDocument) ||
+                type == typeof(MutableJsonElement) ||
+
+                // TODO: support extensible enums?
+                type == typeof(ETag) ||
+
+                // TODO: We'll want to remove this dependency
+                type == typeof(DynamicData) ||
+
+                // TODO: Keep this?
+                type == typeof(object[])
+                ;
+        }
+
+        private static bool IsAllowedPrimitive(Type type)
+        {
+            return
+                type == typeof(bool) ||
+                type == typeof(string) ||
+                type == typeof(byte) ||
+                type == typeof(sbyte) ||
+                type == typeof(short) ||
+                type == typeof(ushort) ||
+                type == typeof(int) ||
+                type == typeof(uint) ||
+                type == typeof(long) ||
+                type == typeof(ulong) ||
+                type == typeof(float) ||
+                type == typeof(double) ||
+                type == typeof(decimal) ||
+                type == typeof(DateTime) ||
+                type == typeof(DateTimeOffset) ||
+                type == typeof(Guid);
+        }
+
+        private static bool IsAllowedArray(Type type)
+        {
+            // TODO: add array support differently
```

Comment:
`Type.IsArray` and `Type.GetElementType()`.

Reply:
TY!